modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js

```diff
@@ -40,8 +40,8 @@ export default async function convertB3dmToI3sGeometry(
 
   const result = [];
   let nodesCounter = nodeId;
-  for (let i = 0; i < (tileContent.gltf.materials.length || 1); i++) {
-    const sourceMaterial = tileContent.gltf.materials[i] || {id: 'default'};
+  for (let i = 0; i < (tileContent.gltf.materials?.length || 1); i++) {
+    const sourceMaterial = tileContent.gltf.materials?.[i] || {id: 'default'};
     if (!convertedAttributesMap.has(sourceMaterial.id)) {
```

Comment:
There is a lot of conditional code here. Logic would be easier to read if separated on multiple lines.
```suggestion
    let {materials = []} = tileContent.gltf;
    if (materials.length === 0) {
      materials.push({id: 'default'};
    }
    
    for (const sourceMaterial of  materials) {
    const sourceMaterial = materials[i];
    if (!convertedAttributesMap.has(sourceMaterial.id)) {
```

Also why do we use a `continue`? Can't we just move the code below into a non-negated if statement?

Reply:
> Also why do we use a continue? Can't we just move the code below into a non-negated if statement?

I remember a case when there is a material that is not linked with some primitive. We have to skip them.

Reply:
> There is a lot of conditional code here. Logic would be easier to read if separated on multiple lines.

Done